torchvision/datasets/core50.py

```diff
@@ -0,0 +1,229 @@
+            training batch order is changed as in the official benchmark.
+        batch (int, optional): One of the training incremental batches from 0 to
+            max-batch - 1. Remember that for the ``ni``, ``nc`` and ``nic`` we
+            have respectively 8, 9 and 79 incremental batches. If
+            ``train=False`` this parameter will be ignored.
+        transform (callable, optional): A function/transform that takes in an
+            PIL image and returns a transformed version. E.g,
+            ``transforms.ToTensor()``
+        target_transform (callable, optional): A function/transform that takes
+            in the target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the
+            internet and puts it in root directory. If dataset is already
+            downloaded, it is not downloaded again.
+
+    Example:
+
+        .. code:: python
+
+            training_data = datasets.CORE50(
+                '~/data/core50', transform=transforms.ToTensor(), download=True
+            )
+            training_loader = torch.utils.data.DataLoader(
+                training_data, batch_size=128, shuffle=True, num_workers=4
+            )
+            test_data = datasets.CORE50(
+                '~/data/core50', transform=transforms.ToTensor(), train=False,
+                download=True
+            )
+            test_loader = torch.utils.data.DataLoader(
+                training_data, batch_size=128, shuffle=True, num_workers=4
+            )
+
+            for batch in training_loader:
+                imgs, labels = batch
+                ...
+
+        This is the simplest way of using the Dataset with the common Train/Test
+        split. If you want to use the benchmark as in the original CORe50 paper
+        (that is for continuous learning) you need to play with the parameters
+        ``scenario``, ``cumul``, ``run`` and ``batch`` hence creating a number
+        of Dataset objects (one for each incremental training batch and one for
+        the test set).
+
+    """
+    ntrain_batch = {
+        'ni': 8,
+        'nc': 9,
+        'nic': 79
+    }
+    urls = {
+        '128x128': 'http://bias.csr.unibo.it/maltoni/download/core50/'
+                   'core50_128x128.zip',
+        '350x350': 'http://bias.csr.unibo.it/maltoni/download/core50/'
+                   'core50_350x350.zip',
+        'filelists': 'https://vlomonaco.github.io/core50/data/'
+                     'batches_filelists.zip'
+    }
+    filenames = {
+        '128x128': 'core50_128x128.zip',
+        '350x350': 'core50_350x350.zip',
+        'filelists': 'batches_filelists.zip'
+    }
+    md5s = {
+        'core50_128x128.zip': '745f3373fed08d69343f1058ee559e13',
+        'core50_350x350.zip': 'e304258739d6cd4b47e19adfa08e7571',
+        'batches_filelists.zip': 'e3297508a8998ba0c99a83d6b36bde62'
+    }
+
+    def __init__(self, root, check_integrity=True, scenario='ni', train=True,
```

Comment:
I don't think we need a `check_integrity` argument in the constructor, and it should by default do the integrity check